tests/components/airly/test_config_flow.py

```diff
@@ -0,0 +1,82 @@
+"""Define tests for the Airly config flow."""
+import pytest
+
+from homeassistant import data_entry_flow
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.components.airly import config_flow
+from homeassistant.components.airly.const import CONF_LANGUAGE, DOMAIN
+
+from tests.common import MockConfigEntry, MockDependency
+
+CONFIG = {
+    CONF_NAME: "abcd",
+    CONF_API_KEY: "foo",
+    CONF_LATITUDE: 0,
+    CONF_LONGITUDE: 0,
+    CONF_LANGUAGE: "en",
+}
+
+
+@pytest.fixture
+def mock_airly():
+    """Mock the airly library."""
+    with MockDependency("airly") as mock_airly_:
+        yield mock_airly_
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    flow = config_flow.AirlyFlowHandler()
+    flow.hass = hass
+
+    result = await flow.async_step_user(user_input=None)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_show_form_with_input(hass):
+    """Test that the form is served with input."""
+    flow = config_flow.AirlyFlowHandler()
+    flow.hass = hass
+
+    result = await flow.async_step_user(user_input=CONFIG)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
```

Comment:
We should test the whole flow to create entry as result, in at least one test. All logic paths in the config flow need to be tested.

Reply:
Here's the current coverage:
https://codecov.io/gh/home-assistant/home-assistant/pull/26375/diff

Reply:
Bump.

Reply:
@MartinHjelmare Could you help me a bit with tests. I have problem with mock depependency. This is [gist](https://gist.github.com/bieniu/18ef65e4a1eec071a1a549bbc334ad02) with tests. I have error:
```
self = <airly.measurements.MeasurementsSession object at 0x7fbe4d07e9b0>

    async def update(self):
        """Get measurements."""
>       data = await self.requests_handler.get(self.request_path)
E       TypeError: object MagicMock can't be used in 'await' expression

venv/lib/python3.6/site-packages/airly/measurements.py:117: TypeError
```
and I don't know how to resolve it.

Reply:
I would expect this test to test what happens after the user has filled the form.

Reply:
@MartinHjelmare test_create_entry tests what happens after the user fass filled the form.

Reply:
Then what are we testing in this test?

Reply:
Indeed, this test is not needed. I will remove it.

Reply:
Done.